jumanji/environments/packing/job_shop/scenario_generator_test.py

```diff
@@ -0,0 +1,147 @@
+        assert_trees_are_equal(scenario1, scenario2)
+
+
+class TestRandomScenarioGenerator:
+    def test_random_scenario_generator__properties(
+        self, random_scenario_generator: RandomScenarioGenerator
+    ) -> None:
+        """Test that the properties of the RandomScenarioGenerator are correct."""
+        assert random_scenario_generator.max_num_jobs == 5
+        assert random_scenario_generator.max_num_ops == 3
+        assert random_scenario_generator.max_op_duration == 4
+
+    def test_random_instance_generator_output_shapes(
+        self, random_scenario_generator: RandomScenarioGenerator
+    ) -> None:
+        """Test that random_instance_generator returns arrays of the expected shapes."""
+
+        num_jobs = 3
+        num_machines = 4
+
+        key = jax.random.PRNGKey(0)
+        scenario = random_scenario_generator(key, num_jobs, num_machines)
+
+        # === Check shapes ===
+
+        assert scenario.ops_machine_ids.shape == (
+            random_scenario_generator.max_num_jobs,
+            random_scenario_generator.max_num_ops,
+        )
+        assert scenario.ops_durations.shape == (
+            random_scenario_generator.max_num_jobs,
+            random_scenario_generator.max_num_ops,
+        )
+        assert scenario.num_ops_per_job.shape == (random_scenario_generator.max_num_jobs,)
+
+        # === Check valid values ===
+
+        # Check valid machine IDs (-1 for padding or valid machine ID)
+        assert jnp.all(
+            (scenario.ops_machine_ids == -1)
+            | ((scenario.ops_machine_ids >= 0) & (scenario.ops_machine_ids < num_machines))
+        )
+
+        # Check valid durations (-1 for padding or positive duration <= max_op_duration)
+        assert jnp.all(
+            (scenario.ops_durations == -1)
+            | (
+                (scenario.ops_durations > 0)
+                & (scenario.ops_durations <= random_scenario_generator.max_op_duration)
+            )
+        )
+
+        # Check valid number of ops per job (between 1 and max_num_ops)
+        assert jnp.all(
+            (scenario.num_ops_per_job >= 0)
+            & (scenario.num_ops_per_job <= random_scenario_generator.max_num_ops)
+        )
+
+        # Check valid number of jobs (equal to num_jobs)
+        assert jnp.sum(jnp.any(scenario.ops_machine_ids != -1, axis=1)) == num_jobs
+
+        # === Check padding ===
+
+        assert jnp.all(scenario.ops_machine_ids[num_jobs:] == -1)
+        assert jnp.all(scenario.ops_durations[num_jobs:] == -1)
+        assert jnp.all(scenario.num_ops_per_job[num_jobs:] == 0)
+
+    def test_random_instance_generator_same_key(
+        self, random_scenario_generator: RandomScenarioGenerator
+    ) -> None:
+        """Validate that the random instance generator's call function is jit-able and compiles
+        only once. Also check that giving two different keys results in two different instances.
```

Comment:
Small mistake in the docstring, it's 2 times the same key gives 2 times the same instance